x-pack/plugins/security/public/components/doc_link.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback, FunctionComponent } from 'react';
+import { EuiLink } from '@elastic/eui';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+import { CoreStart } from '../../../../../src/core/public';
+
+export type DocLinks = CoreStart['docLinks']['links'];
+export type GetDocLinkFunction = (app: string, doc: string) => string;
+
+/**
+ * Creates links to the documentation.
+ *
+ * @see {@link DocLink} for a component that creates a link to the docs.
+ *
+ * @example
+ * ```typescript
+ * <DocLink app="elasticsearch" doc="built-in-roles.html">
+ *   Learn what privileges individual roles grant.
+ * </DocLink>
+ * ```
+ *
+ * @example
+ * ```typescript
+ * const [docs] = useDocLinks();
+ *
+ * <EuiLink href={docs.dashboard.guide} target="_blank" external>
+ *   Learn how to get started with dashboards.
+ * </EuiLink>
+ * ```
+ */
+export function useDocLinks(): [DocLinks, GetDocLinkFunction] {
+  const { services } = useKibana();
+  const { links, ELASTIC_WEBSITE_URL, DOC_LINK_VERSION } = services.docLinks!;
+  const getDocLink = useCallback<GetDocLinkFunction>(
+    (app, doc) => {
+      return `${ELASTIC_WEBSITE_URL}guide/en/${app}/reference/${DOC_LINK_VERSION}/${doc}`;
+    },
+    [ELASTIC_WEBSITE_URL, DOC_LINK_VERSION]
+  );
+  return [links, getDocLink];
+}
+
+export interface DocLinkProps {
+  app: string;
+  doc: string;
+}
+
+export const DocLink: FunctionComponent<DocLinkProps> = ({ app, doc, children }) => {
+  const [, getDocLink] = useDocLinks();
+  return (
+    <EuiLink href={getDocLink(app, doc)} target="_blank" external>
```

Comment:
What we've actually been using here is `<EuiButtonEmpty size="xs" />` so that plugin authors can add icons like to signify going to Github: 
<img width="363" alt="Image 2021-01-04 at 3 18 49 PM" src="https://user-images.githubusercontent.com/549577/103575837-4425e480-4ea0-11eb-8c29-44634beb47ee.png">

This will also help to maintain consistency of sizes.

Reply:
I think the use case is slightly different. The `DocLink` component is specifically created to link to the correct version of the Kibana or Elasticsearch docs at elastic.co. It would not be used to link to Github or other services. 

All the doc links I've seen in Stack Management use the `EuiLink` component with the `external` flag. Happy to change this though if that's not the correct pattern:

<img width="470" alt="Screenshot 2021-01-05 at 11 13 35" src="https://user-images.githubusercontent.com/190132/103639937-1051ca00-4f47-11eb-97b7-973e5fbafa9b.png">

Reply:
Ack, sorry forget my comment then. I saw `DocLink` and just automatically assumed the Help popover.